apps/src/blockly/addons/cdoBlockSvg.js

```diff
@@ -70,20 +70,52 @@ export default class BlockSvg extends GoogleBlockly.BlockSvg {
           Blockly.ContextMenu.hide();
         }.bind(this)
       };
-      const lockToParent = {
-        text: this.canDisconnectFromParent_
-          ? 'Lock to Parent Block'
-          : 'Unlock from Parent Block',
-        enabled: true,
-        callback: function() {
-          this.setCanDisconnectFromParent(!this.canDisconnectFromParent_);
-          Blockly.ContextMenu.hide();
-        }.bind(this)
-      };
       menuOptions.push(deletable);
       menuOptions.push(movable);
       menuOptions.push(editable);
-      menuOptions.push(lockToParent);
+      // menuOptions for Shadow Blocks require awareness of parents and children.
+      const children = this.getChildren();
+      const shadowChildCount = children.filter(child => child.isShadow())
+        .length;
+      const nonShadowChildrenCount = children.filter(child => !child.isShadow())
+        .length;
+      // A block can be made into a shadow if:
+      //* It has a surrounding parent block.
+      //* It does not contain a variable field.
+      //* It does not have any non-shadow children.
+      if (
+        this.getSurroundParent() &&
+        !this.getVarModels().length &&
+        !nonShadowChildrenCount
+      ) {
+        const shadow = {
+          text: 'Make Shadow',
+          enabled: true,
+          callback: function() {
+            this.setShadow(true);
+            Blockly.ContextMenu.hide();
+          }.bind(this)
+        };
+        menuOptions.push(shadow);
+      }
```

Comment:
Ideally I think we should move the logic for determining if a block's context menu should have the shadow option out of this method.

so it would read something like:

```
if (shouldHaveShadowOption()) {
  menuOptions.push(getShadowBlock());
 }
 ```

Reply:
oh actually I didn't realize the other options were declared in this one function as well. so maybe ok to keep  `const shadow = ...` here🤷🏽‍♀️